ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -59,9 +59,13 @@ class GetPrevHash blk => HasAnalysis blk where
     blockHeaderSize  :: blk -> SizeInBytes
     blockTxSizes     :: blk -> [SizeInBytes]
     knownEBBs        :: proxy blk -> Map (HeaderHash blk) (ChainHash blk)
+    protocolInfo     :: [FilePath]
+                     -> Bool       -- is it mainnet?
+                     -> IO (ProtocolInfo IO blk)
     mkTopLevelConfig :: [FilePath] -- Genesis file or files.
                      -> Bool       -- is it mainnet?
                      -> IO (TopLevelConfig blk)
+    mkTopLevelConfig fps onMainNet = pInfoConfig <$> protocolInfo fps onMainNet
```

Comment:
Is this even used?

Reply:
Yes this used by the Main. A little backstory on this: I thought I could use the same api for the db-validator with small modifications. But then I realised I cannot import from executables, only libraries :facepalm: But I think this change makes sense anyway.